buildkite/resource_pipeline.go

```diff
@@ -298,15 +306,16 @@ func CreatePipeline(ctx context.Context, d *schema.ResourceData, m interface{})
 	vars := map[string]interface{}{
 		"cancel_intermediate_builds":               graphql.Boolean(d.Get("cancel_intermediate_builds").(bool)),
 		"cancel_intermediate_builds_branch_filter": graphql.String(d.Get("cancel_intermediate_builds_branch_filter").(string)),
-		"default_branch":                           graphql.String(d.Get("default_branch").(string)),
-		"desc":                                     graphql.String(d.Get("description").(string)),
-		"name":                                     graphql.String(d.Get("name").(string)),
-		"org":                                      orgID,
-		"repository_url":                           graphql.String(d.Get("repository").(string)),
-		"skip_intermediate_builds":                 graphql.Boolean(d.Get("skip_intermediate_builds").(bool)),
-		"skip_intermediate_builds_branch_filter":   graphql.String(d.Get("skip_intermediate_builds_branch_filter").(string)),
-		"steps":                                    graphql.String(d.Get("steps").(string)),
-		"teams":                                    teamsData,
+		"cluster_id":                             graphql.ID(d.Get("cluster_id").(string)),
```

Comment:
The integration tests are failing when this is present and empty. Our GraphQL API doesn't handle this case. If the `cluster_id` key is present in the mutation input, it expects it to be a valid ID.

Are you able to confirm this behaviour if you run a manual test with and without this field in your terraform?

I'm afraid the quickest solution is probably to handle this case in the provider here somehow - possibly duplicating the mutation code and only including `cluster_id` when it its not empty? Or do you have another idea?

Reply:
Ah, thanks for pointing that out.

I ran the manual tests as you suggested (following directions [here](https://github.com/buildkite/terraform-provider-buildkite#manual-testing), and the `terraform plan` works both with and without `cluster_id`. 

I did play with the actual graphql mutation, though and saw what the issue is with passing `cluster_id` when it doesn't exist.

Just pushed up a commit to address that. Lemme know what you think. 🙏

Reply:
That looks like it should do the trick @kate-syberspace. I'll try get it checked thoroughly today.

However, I think we might even be able to fix the mutation from our end to accept null properly (like it should IMO) this week. If we could get a release out with simpler code here would you be happy to wait until early next week?

Reply:
Thanks @jradtilbrook . No rush from my side.  😀